mne_features/feature_extraction.py

```diff
@@ -211,6 +215,89 @@ def _check_func_names(selected, feature_funcs_names):
         return valid_func_names
 
 
+class FeatureExtractor(BaseEstimator, TransformerMixin):
+    """ Feature extraction from epoched EEG data.
+
+    The method `fit_transform` implemented in this class can be used to
+    extract univariate or bivariate features from epoched data
+    (see example below). The method ``fit`` does not have any effect and is
+    implemented for compatibility with Scikit-learn's API. As a result, the
+    class ``FeatureExtractor`` can be used as a step in a Pipeline (see
+    :class:`~sklearn.pipeline.Pipeline` and MNE-features examples).
+
+    Parameters
+    ----------
+    sfreq : float (default: 256.)
+        Sampling rate of the data.
+
+    selected_funcs : list of str or None (default: None)
+        Aliases of the feature functions which will be used to extract
+        features from the data. (See the documentation of mne-features for a
+        complete list of available feature functions).
+
+    params : None
+        If not None, dict of optional parameters to be passed to
+        :func:`extract_features`. Each key of the ``funcs_params`` dict should
+        be of the form: ``[alias_feature_function]__[optional_param]``
+        (for example: ``higuchi_fd__kmax``).
+
+    n_jobs : int (default: 1)
+        Number of CPU cores used when parallelizing the feature extraction.
+        If given a value of -1, all cores are used.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> rng = np.random.RandomState(42)
+    >>> n_epochs, n_channels, n_times = 5, 3, 32
+    >>> X = rng.randn(n_epochs, n_channels, n_times)
+    >>> fe = FeatureExtractor(sfreq=100., selected_funcs=['std', 'kurtosis'])
+    >>> X = fe.fit_transform(X)
+    >>> print(X.shape)
+    (5, 6)
+
+    See also
+    --------
+    :func:`extract_features`
+    """
+    def __init__(self, sfreq=256., selected_funcs=None, params=None, n_jobs=1):
```

Comment:
@jbschiratti you should add a memory parameter to FeatureExtractor to avoid allow caching of computation for functions.

see how it works for example in http://scikit-learn.org/stable/modules/generated/sklearn.pipeline.Pipeline.html

Reply:
For the next PR ?